github/util.py

```diff
@@ -152,40 +153,77 @@ def __init__(self,
         self.to_ref = to_ref
         self.reference_type_name = from_ref.type_name()
 
-    def is_obsolete(self, reference_component):
+    def is_obsolete(
+        self,
+        reference_component: Union[product.model.Component, gci.componentmodel.Component],
+    ):
         '''returns a boolean indicating whether or not this Upgrade PR is "obsolete"
 
         A Upgrade is considered to be obsolete, iff the following conditions hold true:
         - the reference product contains a component reference with the same name
         - the destination version is greater than the greatest reference component version
         '''
         # find matching component versions
-        reference_refs = sorted(
-            [
-                rc for rc in
-                reference_component.dependencies().references(type_name=self.reference_type_name)
-                if rc.name() == self.ref_name
-            ],
-            key=lambda r: version.parse_to_semver(r.version())
-        )
-        if not reference_refs:
-            return False # special case: we have a new reference
+        if isinstance(reference_component, product.model.Component):
```

Comment:
add some "greppable" comment (also for other v1-backard-compat-code) - or create a ticket w/ a link to this piece of code so we do not forget cleanup